### PR TITLE
refactor: use promises for invitation logic

### PR DESCRIPTION
**Pull request by dlozlla**
- refactor pages.js to use promises and improve readability

### CODE REVIEW

**Comment by palmerabollo** [comment]
We are so proud of you...

File: lib/routes/pages.js [hunk 1]
```diff
@@ -47,7 +49,95 @@ function pickEmail(emailsObj) {
...
+  return new Promise((resolve, reject) => {
+    var validEmail = pickEmail(emails);
+    if (validEmail !== null) {
+      resolve(validEmail);
+    } else {
+      logger.info('Github user %s has been rejected to join the organization', username);
+      var message = 'SORRY, you cannot join our organization..., this is nothing personal, just work. To gain access, ' +
+      'your github profile needs to include a validated email address, belonging to a company within ' +
+      'the Telefonica ecosystem';
+      reject(message);
```
Comment (palmerabollo) [inline lib/routes/pages.js]: It works but we usually reject with an Error. Example: `reject(new Error(message))`

File: lib/routes/pages.js [hunk 2]
```diff
@@ -47,7 +49,95 @@ function pickEmail(emailsObj) {
...
+      var message = 'CONGRATS!! we have invited you to our organization. ' +
+                'Accept the invitation at github and remember ' +
+                'to <a href="https://github.com/settings/security">enable two-factor auth</a> ' +
+                'to protect the code.';
+      return message;
+    })
+    .catch((error) => {
+      //if there is an error when inviting user, log the error, undo the changes in database and let the user know
+      logger.error('An unexpected error occurred while inviting github user %s: %s', username, error);
+      var q = {};
```
Comment (palmerabollo) [inline lib/routes/pages.js]: with ES6 you can do `var q = { username }`

File: lib/routes/pages.js [hunk 3]
```diff
@@ -58,112 +148,35 @@ function authError(req, res) {
...
-  logger.debug('Github user %s with AccessToken %s is requesting to join', req.user.username, req.user.accessToken);
-
-  github.user.getEmails({}, function(userErr, userRes) {
-    if (userErr) {
-      logger.error('An unexpected error occurred while using github API to get github user %s email addresses: %s ',
-          req.user.username, userErr);
+  logger.info('Github user %s is requesting to join', req.user.username);
+
+  github.users.getEmails({})
+    //check the user fulfills the criteria to be invited
```
Comment (palmerabollo) [inline lib/routes/pages.js]: The bad part of anonymous functions is that you can not use the function name to avoid these kind of comments

Comment (dlozlla) [inline lib/routes/pages.js]: maybe the comment is really not needed, as you can read "validateUser" within the then(). Perhaps I was too verbose in the comments

**Pull request merged**